### PR TITLE
blocks seive v2

### PR DESCRIPTION
**Pull request by JuliaGerasymenko**

### CODE REVIEW

File: oneMoreSeive.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+function seiveBlocks(n) {
+  let primes = new Array();
+  let nPrimes = new Array(n + 1).fill(false);
```
Comment (DzyubSpirit) [inline oneMoreSeive.js:5]:
```suggestion
  let nPrimes = new Array(Math.floor(Math.sqrt(n)) + 1).fill(false);
```

File: oneMoreSeive.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      for (let j = i * i; j <= Math.sqrt(n); j += i) {
+        nPrimes[j] = true;
+      }
+    }
+  }
+
+  let blockSize = 3;
+  for (let k = 0; k < Math.ceil(n / blockSize); k++) {
+    for (let i = 2; i < blockSize + 2; i++) {
+      for (let prime = 0; prime < primes.length; prime++) {
```
Comment (DzyubSpirit) [inline oneMoreSeive.js:20]:
```suggestion
      for (let prime of primes) {
```

**Review by DzyubSpirit — COMMENTED** [review]
You're counting the sum of the primes number, right?
Just clarifying, it doesn't really matter because the main algorithm are the same.

File: oneMoreSeive.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  let result = 0;
+  nPrimes.forEach((el, i) => {
+    if (!el) result += i;
+  });
+  return result;
+}
+
+console.log(seiveBlocks(6));
```
Comment (DzyubSpirit) [inline oneMoreSeive.js:42]: Try to test with more cases. Find the edge cases for your code and put them here.

File: oneMoreSeive.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      for (let prime = 0; prime < primes.length; prime++) {
+        if (k * blockSize + i <= n && !nPrimes[k * blockSize + i]) {
+          let compare = k * blockSize + i - 1;
+          let p =
+            compare >= blockSize - 1
+              ? Math.floor(compare / primes[prime]) + 1
+              : i;
+          p = p * primes[prime];
+          for (; p <= blockSize * (k + 1) + 1; p += primes[prime])
+            nPrimes[p] = true;
```
Comment (DzyubSpirit) [inline oneMoreSeive.js:29]: You're not supposed to use n-length array here. Try to find a solution without it.

File: oneMoreSeive.js (hunk 1 above)
Comment (JuliaGerasymenko) [reply oneMoreSeive.js:5]: I use nPrimes later to cross out complex numbers.

Comment (DzyubSpirit) [reply oneMoreSeive.js:5]: I know. But the point of the algorithm is to not store n-length array otherwise it would be possible to just use the standard Sieve of Eratosthenes
